File: Lab2.1/AlphaString.cpp
#include "AlphaString.h"
#include <algorithm>

AlphaString::AlphaString() : String() {}

AlphaString::AlphaString(const std::string& val) : String(val) {
    if (!isAlpha()) {
        value = "";
    }
}

AlphaString::AlphaString(const AlphaString& other) : String(other) {}

void AlphaString::shiftRight() {
    if (value.empty()) return;
    char last = value.back();
    value.pop_back();
    value = last + value;
}

bool AlphaString::isAlpha() const {
    return std::all_of(value.begin(), value.end(), [](char c) {
        return std::isalpha(c);
        });
}

File: Lab2.1C#/AlphaString.cs
using System;
using System.Linq;

namespace StringLibrary
{
    public class AlphaString : String
    {
        public AlphaString() : base() { }

        public AlphaString(string val) : base(val)
        {
            if (!IsAlpha())
                value = "";
        }

        public AlphaString(AlphaString other) : base(other) { }

        public void ShiftRight()
        {
            if (value.Length == 0) return;
            value = value[^1..] + value[..^1];
        }

        private bool IsAlpha()
        {
            return value.All(char.IsLetter);
        }
    }
}